randovania/games/fusion/gui/dialog/game_export_dialog.py

```diff
@@ -25,19 +29,69 @@ class FusionGameExportDialog(GameExportDialog, Ui_FusionGameExportDialog):
     def game_enum(cls) -> RandovaniaGame:
         return RandovaniaGame.FUSION
 
+    def __init__(self, options: Options, patch_data: dict, word_hash: str, spoiler: bool, games: list[RandovaniaGame]):
+        super().__init__(options, patch_data, word_hash, spoiler, games)
+
+        self._base_output_name = f"MARS - {word_hash}"
+        per_game = options.options_for_game(self.game_enum())
+        assert isinstance(per_game, FusionPerGameOptions)
+
+        # Input
+        self.input_file_button.clicked.connect(self._on_input_file_button)
+
+        # Output
+        self.output_file_button.clicked.connect(self._on_output_file_button)
+
+        if per_game.input_path is not None:
+            self.input_file_edit.setText(str(per_game.input_path))
+
+        if per_game.output_path is not None:
+            output_path = per_game.output_path.joinpath(self.default_output_name)
+            self.output_file_edit.setText(str(output_path))
+
+        add_field_validation(
+            accept_button=self.accept_button,
+            fields={
+                self.input_file_edit: lambda: False,
+                self.output_file_edit: lambda: False,
+            },
+        )
+
+    @property
+    def valid_input_file_types(self) -> list[str]:
+        return ["gba"]
+
+    @property
+    def valid_output_file_types(self) -> list[str]:
+        return ["gba"]
+
     @property
     def input_file(self) -> Path:
-        raise NotImplementedError("This method hasn't been implemented yet")
+        return Path(self.input_file_edit.text())
 
     @property
     def output_file(self) -> Path:
-        raise NotImplementedError("This method hasn't been implemented yet")
+        return Path(self.output_file_edit.text())
 
     @property
     def auto_save_spoiler(self) -> bool:
-        raise NotImplementedError("This method hasn't been implemented yet")
+        return self.auto_save_spoiler_check.isChecked()
+
+    # Input file
+    def _on_input_file_button(self):
+        input_file = prompt_for_input_file(self, self.input_file_edit, self.valid_input_file_types)
+        if input_file is not None:
+            self.input_file_edit.setText(str(input_file.absolute()))
+
+    # Output File
+    def _on_output_file_button(self):
+        output_file = prompt_for_output_file(
+            self, self.valid_output_file_types, self._base_output_name, self.output_file_edit
+        )
+        if output_file is not None:
+            self.output_file_edit.setText(str(output_file))
 
-    def update_per_game_options(self, per_game: PerGameOptions) -> FusionPerGameOptions:
+    def update_per_game_options(self, per_game: FusionPerGameOptions) -> FusionPerGameOptions:
```

Comment:
this should be PerGameOptions.

Reply:
I referenced AM2R and Prime and they both had their respective game options

Reply:
interesting. pretty sure that mypy should complain about that